### PR TITLE
Promoters feature

### PR DESCRIPTION
**Pull request by raularango85**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -13,10 +13,11 @@ REACT_APP_PREVIEW_REACT_URL=https://preview.keenthemes.com/metronic8/react
 REACT_APP_PREVIEW_DOCS_URL=https://preview.keenthemes.com/metronic8/react/docs
 REACT_APP_THEME_API_URL=https://preview.keenthemes.com/theme-api/api
 
-REACT_APP_PUBLIC_URL=http://localhost:3011
```
Comment (dannytp91) [inline .env:16]: Avoid change this file

File: src/_metronic/layout/components/sidebar/SidebarLogo.tsx [hunk 2]
```diff
@@ -16,13 +18,17 @@ const SidebarLogo = () => {
     : ''
   const toggleState = appSidebarDefaultMinimizeDesktopEnabled ? 'active' : ''
   const appSidebarDefaultMinimizeDefault = config.app?.sidebar?.default?.minimize?.desktop?.default
+  const logoPath = promoterInfo.promoterLogo
+    ? process.env.REACT_APP_USERS_BASE_URL?.replace(/\\/g, '/') + promoterInfo.promoterLogo
```
Comment (dannytp91) [inline src/_metronic/layout/components/sidebar/SidebarLogo.tsx]: use template string

File: src/_metronic/layout/components/sidebar/sidebar-menu/SidebarMenuMain.tsx [hunk 3]
```diff
@@ -7,9 +7,10 @@ import {useProtectedRoute} from '../../../../../app/components/RouteProtection/P
 
 const SidebarMenuMain = () => {
   const intl = useIntl()
-  const allRolesRoutes = useProtectedRoute(['Admin', 'Scrutiny'])
+  const allRolesRoutes = useProtectedRoute(['Admin', 'Scrutiny', 'Promoter'])
```
Comment (dannytp91) [inline src/_metronic/layout/components/sidebar/sidebar-menu/SidebarMenuMain.tsx]: it is good create a constant that contains roles string, and create a new constants creating a array with constants of roles

File: src/_metronic/layout/components/sidebar/sidebar-menu/SidebarMenuMain.tsx [hunk 4]
```diff
@@ -7,9 +7,10 @@ import {useProtectedRoute} from '../../../../../app/components/RouteProtection/P
...
   const adminScrutinyRoutes = useProtectedRoute(['Admin', 'Scrutiny'])
   const adminRoutes = useProtectedRoute(['Admin'])
```
Comment (dannytp91) [inline src/_metronic/layout/components/sidebar/sidebar-menu/SidebarMenuMain.tsx]: use constants for roles

File: src/app/components/Tables/ScrutinySettings/EmailScrutinySettingsTable.tsx [hunk 5]
```diff
@@ -33,38 +33,46 @@ const ScrutinySettingsTable = ({
...
+                    <td className='text-center'>
+                      <div className='d-flex align-items-center justify-content-center'>
+                        <div
+                          onClick={() => setEmailId(email.adminEmailId, 'update')}
+                          style={{cursor: 'pointer', marginRight: '10px'}}
+                        >
+                          <i className='bi bi-pencil text-primary btn-lg'></i>
+                        </div>
+                        <div
+                          onClick={() => setEmailId(email.adminEmailId, 'delete')}
```
Comment (dannytp91) [inline src/app/components/Tables/ScrutinySettings/EmailScrutinySettingsTable.tsx]: please create a function outside of on click, create inline anonymous function is not a good practice, and sometimes those functions could create side effects or not execute

**Review by dannytp91 — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (raularango85) [reply .env:16]: .env file added to .gitignore file... be careful because .env file is in delete mode

**Review by dannytp91 — APPROVED** [review]

**Pull request merged**